examples/navier-stokes/nsplex.c

```diff
@@ -868,6 +869,31 @@ int main(int argc, char **argv) {
     if (qf_ifunction) CeedQFunctionSetContext(qf_ifunction, &ctxAdvection2d, sizeof ctxAdvection2d);
   }
 
+  // Copied from ../petsc/bps.c
+  // Create the error Q-function
+  CeedQFunction qf_error;
+  CeedOperator op_error;
+  CeedVector target;
+  CeedQFunctionCreateInterior(ceed, 1, Error,
+                              Error_loc, &qf_error);
+  CeedQFunctionAddInput(qf_error, "q", ncompq, CEED_EVAL_INTERP);
+  CeedQFunctionAddInput(qf_error, "true_soln", ncompq, CEED_EVAL_NONE);
```

Comment:
The true solution depends on time, which we'll pass in through the context. One way to leverage code is to add `time` to the context for initial conditions, and have that error when `time != 0.0` unless it's supported. In that case, you'd call initial conditions with this new time and collocated output to define the exact solution, then use this differencing to make it an error, also collocated at quadrature points.

Reply:
What do you mean when you say the "context"? Is that a struct somewhere?

Reply:
See `CeedQFunctionSetContext`. It's how we pass in parameters like viscosity and gravitational acceleration.